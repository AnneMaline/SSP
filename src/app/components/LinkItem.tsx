import React from "react";

type LinkItemProps = {
  title: string;
  information: string;
  tags: string[];
  link: string;
};

const LinkItem = ({ title, information, tags, link }: LinkItemProps) => {
  return (
    <div className="py-4 border-b border-gray-200">
      <h2 className="text-lg font-bold">{title}</h2>
      <p className="text-gray-700 my-2">{information}</p>
      <div className="mb-2">
        {tags.map((tag, index) => (
          <span
            key={index}
            className="inline-block bg-blue-100 text-blue-800 text-sm font-medium mr-2 px-2.5 py-0.5 rounded"
          >
            {tag}
          </span>
        ))}
      </div>
      <a
        href={link}
        target="_blank"
        rel="noopener noreferrer"
        className="text-blue-500 hover:underline"
      >
        Learn More
      </a>
    </div>
  );
};

export default LinkItem;
