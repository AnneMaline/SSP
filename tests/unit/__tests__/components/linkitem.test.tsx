import { render } from "@testing-library/react";
import LinkItem from "@/components/LinkItem";

describe("StepCard", () => {
  const mockProps = {
    title: "Test Title",
    information: "Test information about the item.",
    tags: ["Tag1", "Tag2", "Tag3"],
    link: "https://example.com",
  };
  it("renders correctly", () => {
    const { container } = render(<LinkItem {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
