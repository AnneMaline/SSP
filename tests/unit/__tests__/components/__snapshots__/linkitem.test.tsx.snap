// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepCard renders correctly 1`] = `
<div>
  <div
    class="py-4 border-b border-gray-200"
  >
    <h2
      class="text-lg font-bold"
    >
      Test Title
    </h2>
    <p
      class="text-gray-700 my-2"
    >
      Test information about the item.
    </p>
    <div
      class="mb-2"
    >
      <span
        class="inline-block bg-blue-100 text-blue-800 text-sm font-medium mr-2 px-2.5 py-0.5 rounded"
      >
        Tag1
      </span>
      <span
        class="inline-block bg-blue-100 text-blue-800 text-sm font-medium mr-2 px-2.5 py-0.5 rounded"
      >
        Tag2
      </span>
      <span
        class="inline-block bg-blue-100 text-blue-800 text-sm font-medium mr-2 px-2.5 py-0.5 rounded"
      >
        Tag3
      </span>
    </div>
    <a
      class="text-blue-500 hover:underline"
      href="https://example.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn More
    </a>
  </div>
</div>
`;
